File: kart-dashboard-backend/src/database/logger.ts
class Logger {
    private classname: string;

    constructor(classsname: string) {
        this.classname = classsname;
    }

    // Log time
    private logWithTime(): string {
        const currentDate = new Date();

        const day = String(currentDate.getDate()).padStart(2, '0');
        const month = String(currentDate.getMonth() + 1).padStart(2, '0');
        const year = currentDate.getFullYear();

        const hours = String(currentDate.getHours()).padStart(2, '0');
        const minutes = String(currentDate.getMinutes()).padStart(2, '0');

        return `${day}-${month}-${year} ${hours}:${minutes}`;
    }

    // Log info
    public Info(message: string) {
        this.Log(message, "INFO");
    }

    // log warning
    public Warning(message: string) {
        this.Log(message, "WARNING");
    }

    // log debug
    public Debug(message: string) {
        this.Log(message, "DEBUG");
    }

    // error
    public Error(message: string) {
        this.Log(message, "ERROR");
    }

    // Log message with classname, message and level
    public Log(message: string, level: string) {
        console.log(`[${this.logWithTime()} - ${this.classname} - ${level}]: ${message}\n`);
    }
}

export default Logger;

File: kart-dashboard-backend-rewrite/src/components/logger.rs
use chrono::{DateTime, Local};

pub struct Logger {
    structname: String,
}

#[allow(dead_code)]
impl Logger {
    #[allow(dead_code)]
    // Constructor
    pub fn new(structname: &str) -> Self {
        let structname = structname.to_string();
        Logger { structname }
    }

    // Log with time
    fn log_with_time(&self) -> String {
        let current_time: DateTime<Local> = Local::now();
        let formatted_time = current_time.format("%d-%m-%Y %H:%M:%S").to_string();
        formatted_time
    }

    // Print out the message, level, and structname
    fn log(&self, level: &str, message: &str) {
        let level: String = level.to_string();
        let message: String = message.to_string();
        let time: String = self.log_with_time();
        println!("[{} {} - {}]: {}", time, self.structname, level, message)
    }

    // Warnings
    pub fn warning(&self, message: &str) {
        self.log("WARNING", message);
    }

    // Errors
    pub fn error(&self, message: &str) {
        self.log("ERROR", message);
    }

    // Info
    pub fn info(&self, message: &str) {
        self.log("INFO", message);
    }

    // Debug
    pub fn debug(&self, message: &str) {
        self.log("DEBUG", message);
    }
}
